docs: jcloud docs revamp for instance (#5759)
jina-ai/jina: docs/concepts/jcloud/yaml-spec.md

JCloud extends Jina's {ref}`Flow YAML specification<flow-yaml-spec>` by introducing the special field `jcloud`. This lets you define resources and scaling policies for each Executor and gateway.

> Here's a Flow with two Executors that have specific resource needs. `indexer` demands 10G `ebs` disk, whereas `encoder` demands G5 instance, which implies 4 cores and 8G RAM are used. See below sections for further information about Instance Type.

Reviewer: ```suggestion
Here's a Flow with two Executors that have specific resource needs: `indexer` requires a 10 GB `ebs` disk, whereas `encoder` requires a G5 instance, which implies that four cores and 8 GB RAM are used. See the below sections for further information about Instance Types.
```

Reviewer: Let's use [Wikipedia abbreviation conventions](https://en.wikipedia.org/wiki/Gigabyte) for consistency

| C4       | 2     | 4 GB     | 20               |
| C5       | 4     | 8 GB     | 40               |
| C6       | 8     | 16 GB    | 80               |
| C7       | 16    | 32 GB    | 160              |
| C8       | 32    | 64 GB    | 320              |

By default, C1 is allocated to each Executor and Gateway.

JCloud offers the general Intel Xeon processor (Skylake 8175M or Cascade Lake 8259CL) for the CPU instances. 

#### GPU Tiers

JCloud supports GPU workloads with two different usages: `shared` or `dedicated`. 

If GPU is enabled, JCloud will provide NVIDIA A10G Tensor Core GPUs with 24G memory for workloads in both usage types.

```{hint}
When using GPU resources, it may take a few extra minutes before all Executors are ready to serve traffic.
```

| Instance | GPU    | Memory   | Credits Per Hour |
|----------|--------|----------|------------------|
| G1       | shared | 14 GB    | 100              |
| G2       | 1      | 14 GB    | 125              |
| G3       | 2      | 24 GB    | 250              |
| G4       | 4      | 56 GB    | 500              |

##### Shared GPU

An Executor using a `shared` GPU shares this GPU with up to four other Executors.

> This enables time-slicing, which allows workloads that land on oversubscribed GPUs to interleave with one another.

Reviewer 2: ```suggestion
This enables time-slicing, which allows workloads that land on oversubscribed GPUs to interleave with one another. 
To use `shared` GPU, `G1` needs to be specified as instance type.
```

-> move this sentence up

Author: done

The tradeoffs with a `shared` GPU are increased latency, jitter, and potential out-of-memory (OOM) conditions when many different applications are time-slicing on the GPU. If your application is memory consuming, we suggest using a dedicated GPU.

> To use `shared` GPU, `G1` needs to be specified as instance type.

Reviewer 2: ```suggestion
```
move sentence up. It is now between two blocks that talk about dedicated GPUs

Author: done

In JCloud, you can pass highly customizable, finely-grained resource requests for each Executor using the `jcloud.resources` argument in your Flow YAML.

### Instance

> JCloud uses the concept of "Instance" to respresent the hardware specifications. 

Reviewer: ```suggestion
JCloud uses the concept of an "Instance" to represent a specific set of hardware specifications. 
```
Is this correct?

Also, please spell check next time :sweat_smile:

Author: Yea that's correct, oops sorry I should have installed the spelling checker in the IDE.

Author: fixed

> In above example, C2 Instance Type stands for 4 cores and 8G RAM based on the CPU Tiers Instance defination table below.

Reviewer: ```suggestion
In above example, a C2 Instance Type represents four cores and 8 GB RAM based on the CPU Tiers Instance definition table below.
```

Please:

- Spell check throughout
- Switch `8G` to `8 GB` (with space) throughout

Author: cool fixed

Note that if you are still using the legacy resource specification interface, as such:

```{code-block} yaml
jcloud:
  resources:
    cpu: 8
    memory: 8G
```

We would translate the raw numbers from input to Instance tier that fits most. 
There are circumstances the Instance tier does not exactly fufill the CPU cores and Memory you need, like in above exmaple. 
But we would "ceil" the requests to the lowest tier that would satisfy all the specifications. 
In this case, `C6` would be consisdered, as `C5`'s `Cores` is lower than what's being requested (4 vs 8). 

There are also two types of Instance Tiers, one for CPU instances, one for GPU.

#### CPU Tiers

> | Instance | Cores | Memory | Credits Per Hour |

Reviewer 2: We introduce `Credits Per Hour` in this table, but do not explain where this is coming from or what they mean. I suggest we introduce the credits concept and/or link to the pricing page.

https://cloud.jina.ai/pricing

Author: Yea that's definitely lacking, I will add this piece of info in!

Reviewer: I'll check this again after you add that info @zac-li

Author: added some

When using GPU resources, it may take a few extra minutes before all Executors are ready to serve traffic.
```

> | Instance | GPU    | Memory | Credit Per Hour |

Reviewer: ```suggestion
| Instance | GPU    | Memory | Credits per hour |
```

Sentence case for table headers throughout please

|----------|--------|--------|-----------------|
| G1       | shared | 14G    | 100             |
| G2       | 1      | 14G    | 125             |
| G3       | 2      | 24G    | 250             |

> | G4       | 4      | 56G    | 500             |

Reviewer: Fix column 3 for `<number> GB`

##### Shared GPU

An Executor using a `shared` GPU shares this GPU with up to four other Executors.
This enables time-slicing, which allows workloads that land on oversubscribed GPUs to interleave with one another.

The tradeoffs with a `shared` GPU are increased latency, jitter, and potential out-of-memory (OOM) conditions when many different applications are time-slicing on the GPU. If your application is memory consuming, we suggest using a dedicated GPU.

> To use `shared` GPU, `G1` needs to be specificied as Instance Type.

Reviewer: ```suggestion
To use `shared` GPU, `G1` needs to be specificied as Instance Type.
```
Why are we capitalizing `Instance Type`?

Author: I just thought it's a special term we come up with, or you think it should be`Instance type` or just all lower cased `instance type`?

Reviewer: Which of us came up with it? If no compelling reason otherwise, I'd go with `instance type` (all lower case)

Author: I did, yea no compelling reason, switched to all lower case.

##### Dedicated GPU

> Using a dedicated GPU is the default way to provision GPU for an Executor. This automatically creates nodes or assigns the Executor to land on a GPU node. In this case, the Executor owns the whole GPU. 

Reviewer: ```suggestion
Using a dedicated GPU is the default way to provision a GPU for an Executor. This automatically creates nodes or assigns the Executor to land on a GPU node. In this case, the Executor owns the whole GPU.
```

What do you mean by `land`?

Author: I'm not exactly sure, let me check with the team.

Author: there's nothing special to it, maybe we can just do:

```
assigns the Executor to a GPU node
```

?

Reviewer: Yup, that sounds better

Author: fixed
